### ISSUE TITLE
[GetConnect] Error when my API no return contentType

### ISSUE THREAD

**Issue by jasonlaw** [jonataslaw/getx#884]
**ATTENTION: DO NOT USE THIS FIELD TO ASK SUPPORT QUESTIONS. USE THE PLATFORM CHANNELS FOR THIS. THIS SPACE IS DEDICATED ONLY FOR BUGS DESCRIPTION.**
**Fill in the template. Issues that do not respect the model will be closed.**

**Describe the bug**
When my API has void return type, GetConnect will hit the following error.
~~~dart
[GETX] OPEN DIALOG: 495373635
[GETX] CLOSE DIALOG: 495373635
E/flutter ( 6727): [ERROR:flutter/lib/ui/ui_dart_state.cc(177)] Unhandled Exception: NoSuchMethodError: The getter 'mimeType' was called on null.
E/flutter ( 6727): Receiver: null
E/flutter ( 6727): Tried calling: mimeType
E/flutter ( 6727):
~~~

Here is my API in ASP.NET.
~~~C#
 [HttpPost, Route("{key}/delete")]
 public void DeleteDiscussion(Guid key)
~~~

**_Please notes that the API is executed successfully, so the exception should be raised upon processing the response from API._**

**Flutter Version:**
1.22.4

**Getx Version:**
3.22.2

**Describe on which device you found the bug:**
ex: Emulator - Android.

**Comment by brunozupp** [comment]
I have faced the same error in the project I'm working. I have two scenarios from status code that I can receive from API (NotFound and Accepted). When I log into the app with an user that doesn't have a customer yet, I receive the same error from the issues title and the status code 400 (Bad Request), but I should receive status code Not Found in this case.

```
Exception has occurred.
NoSuchMethodError (NoSuchMethodError: The getter 'mimeType' was called on null.
Receiver: null
Tried calling: mimeType)
```

If I log into the app with an user that already has registered a customer, it returns correctly and I get the status code Accepted. The code below is about my requision to the API

```
Future<Response<Customer>> httpGetByToken({@required String token}) async {
    
    Map<String, String> headers = {
      HttpHeaders.authorizationHeader: 'Bearer $token',
    };

    final response = await get(
      url_customer_get,
      headers: headers,
      contentType: 'application/json'
    ).catchError((_) => Response<Customer>(
      statusCode: HttpStatus.badRequest, 
      statusText: "Erro na operação"
    ));

    // First access, I don't have a customer registered in the user's account yet.
    if (response.statusCode == HttpStatus.notFound) {
      return Response<Customer>(
        statusCode: response.statusCode,
        statusText: "Não encontrado nenhum customer com a identificação do usuário"
      );
    }

    // User has already registered a customer.
    if (response.statusCode == HttpStatus.accepted) {
      return Response<Customer>(
        statusCode: response.statusCode,
        body: Customer.fromJson(response.bodyString)
      );
    }

    // If something wrong happens, return this.
    return Response<Customer>(
      statusCode: response.statusCode,
      statusText: "Erro na operação"
    );
  }
```

To summarize, in the api the code Not Found is sent to me without content and the code Accepted is sent to me with the registered customer in the content from requision. 

I tried some things like taking off the generics <Customer> and putting the code above inside a try/catch, but no result, I've continued receiving the same error.

**Flutter version: 1.22.5

Emulator: Android

GetX version:  3.22.2**

**Comment by gabrimatic** [comment]
Same issue here!

I wrote a code to check if the token is unauthorized, refresh it.
```
if (response.statusCode == HttpStatus.unauthorized)
      await _refreshToken();
```

but I get this exception:

> NoSuchMethodError: The getter 'mimeType' was called on null


**Flutter version: 1.22.5
Emulator: Android
GetX version: 3.22.2**
